contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -213,27 +280,46 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
 
         (address recipient, uint256 amount) = abi.decode(message, (address, uint256));
         require(recipient != address(0), "NativeTokenDestination: zero recipient address");
-        require(amount != 0, "NativeTokenDestination: zero transfer value");
+
+        uint256 scaledAmount = _scaleTokens(amount, true);
 
         // If the contract has not yet been collateralized, we will deduct as many tokens
         // as needed from the transfer as needed. If there are any excess tokens, they will
         // be minted and sent to the recipient.
-        uint256 adjustedAmount = amount;
+        uint256 adjustedAmount = scaledAmount;
```

Comment:
Lets read `currentReserveImbalance` into a local variable here so that we don't read it from storage 3+ times. Ideally it would be one read and at most one write.